README.md

```diff
@@ -17,13 +17,37 @@ xsd2.exe &lt;schema file&gt; [/o:&lt;output-directory&gt;] [/ns:&lt;namespace&gt
 
 ### Example running for embedding in your CSPROJ (C# project):
 
-	<ItemGroup>
-		<XSDFile Include="**/*.xsd" />  
-	</ItemGroup>  
-	<Target Name=“GenerateSerializationClasses” BeforeTargets=“BeforeBuild” Inputs="%(XSDFile.Identity)" Outputs="%(XSDFile.Identity).cs">  
-		<Message Importance=“High” Text=“Generating xsd code…%(XSDFile.Identity)” />  
-		<Exec Command=“xsd2.exe %(XSDFile.Identity) /o:$(ProjectDir) /ns:My.Namespace.Example” />  
-	</Target>
+```xml
+  <PropertyGroup>
+    <!-- XsdFilesPath points here to currelt folder. It can poing for example to external folder, like ..\..\sdk\schema\ -->
+    <XsdFilesPath>.\</XsdFilesPath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Include="$(XsdFilesPath)**\*.xsd" />
+  </ItemGroup>
+  <ItemGroup>
+    <XSDFile Include="$(XsdFilesPath)**\*.xsd" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Update="*.cs">
+      <DependentUpon>$(XsdFilesPath)$([System.String]::Copy('%(Filename)').Split('.')[0]).xsd</DependentUpon>
```

Comment:
Does this work for files like `This.That.cs` that depends on `This.That.xsd`? Maybe there is a better way to change the file extensions in MSBuild? Maybe you can use `System.IO.Path.ChangeExtension`?

Reply:
I think not.
This is just an example as I wrote, of course.
For cases more than one dot, there should be a differen solution
Maybe to use Parh extension methods.
Or even it might not needed to use that split at all.I haven't tried it, but the filename might be without the extension.
The split code example was taken from a code like 'name.gen.cs' which the 'gen' should have been ignored.

Tou can try removing that split and it might do the job for what you like.

I can test it some day as well.

Reply:
I have tested and **pushed** now a new commit.

the **System.Copy** and **Split** weren't necessary at all.
You can see the new pushed example with a much simpler code.

Anyway, this "**DependentUpon**" is not related to the file changes detection.
This is only to make VS folding the generated file under its source - the xsd generated it - in the solution view.

As for the file changes detection and regenerating only what/if being changed, I have learned today how to do it correctly:

1. To make VS watch the files, I took the idea of **UpToDateCheckInput Include="file.dat"** mentioned in [this StackOverflow discussion](https://stackoverflow.com/questions/50026355/how-to-avoid-that-visual-studio-incremental-build-does-not-run-when-files-outsid).
2. The matching between Inputs and Outputs can be done pefectly using [MSBuild transforms](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-transforms?view=vs-2019). Example is also in [How to: Build incrementally](https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-build-incrementally?view=vs-2019)

I hope the new example will answer your needs.